services/scan/src/precinct_scanner_state_machine.ts

```diff
@@ -476,10 +487,20 @@ function buildMachine(createPlustekClient: CreatePlustekClient) {
         on: {
           WAIT_FOR_PAPER: 'no_paper',
           SCANNER_NO_PAPER: { target: 'accepted', internal: true },
-          SCANNER_READY_TO_SCAN: {
-            target: 'ready_to_scan',
-            actions: clearLastScan,
+          SCANNER_READY_TO_SCAN: { target: 'accepted', internal: true },
+        },
+        // TODO should this timeout be controlled by the frontend
+        after: { 3000: 'accepted_and_waiting_for_paper' },
```

Comment:
@benadida I'm curious to hear your thoughts on the best way to implement this. For the 5s timeout to move from `accepted` back to `no_paper`, I added the `WAIT_FOR_PAPER` command so the frontend could control when that happens. Now, we have three cases after accept:
- (immediately) don't scan new ballots, show accepted screen
- (3s later) scan new ballots, show accepted screen
- (5s later) scan new ballots, show insert ballot screen

Should both transitions be controlled by the frontend? Or just one? Or neither?

I'm starting to think maybe these could just be config for the state machine rather than frontend concerns. Seems like a bit of overkill to have multiple API calls for these transitions when the delay will always be the same. Plus, there's no architectural reason the frontend needs to be in control of this (i.e. if an admin card was inserted before one of these delays, nothing bad would happen)

Reply:
I like your approach of putting all of these in the state machine. The only exception is the actual decision to start scanning a ballot, which I think you moved to the frontend for permissioning sanity, which makes sense. I would not move anything more to the front-end, I would add other states, as you suggest, for these things.